executor/write.go

```diff
@@ -200,21 +200,27 @@ func (e *DeleteExec) deleteMultiTables() error {
 			break
 		}
 
-		for _, entry := range joinedRow.RowKeys {
-			if !isMatchTableName(entry, tblMap) {
-				continue
-			}
-			if tblRowMap[entry.Tbl] == nil {
-				tblRowMap[entry.Tbl] = make(map[int64][]types.Datum)
+		for id, cols := range e.SelectExec.Schema().TblID2handle {
+			tbl := e.tblID2Table[id]
+			for _, col := range cols {
+				if !isMatchTableNameNew(id, col, tblMap) {
+					continue
+				}
+				if tblRowMap[id] == nil {
+					tblRowMap[id] = make(map[int64][]types.Datum)
+				}
+				offset := getTableOffset(e.SelectExec.Schema(), col)
+				end := offset + len(tbl.Cols())
+				data := joinedRow.Data[offset:end]
+				handle := joinedRow.Data[col.Index].GetInt64()
+				tblRowMap[id][handle] = data
```

Comment:
No, such as `delete from t t1, t t2 where t1.a > 1 and t2.a < -1`, we should store data from two datasource not only one.